src/buffer_manager.ts

```diff
@@ -442,6 +443,15 @@ export class BufferManager implements Disposable {
         );
     }
 
+    // #247
+    private handleBufferModifiedSet({ buf, modified }: EventBusData<"BufModifiedSet">[0]) {
+        if (modified) return; // This was received because we modified the buffer; this is expected.
+        const doc = this.getTextDocumentForBufferId(buf);
+        if (doc && doc.isDirty && !doc.isUntitled && !doc.isClosed) {
+            doc.save();
```

Comment:
@xiyaowong Is it possible to save without formatting? (`workbench.action.files.saveWithoutFormatting`)

Reply:
Easily achievable by editing format on save setting

Reply:
> @xiyaowong Is it possible to save without formatting? (`workbench.action.files.saveWithoutFormatting

You can only modify your formatOnSave settings. Additionally, since it is an undo operation and automatic formatting is enabled, the content and formatting should meet expectations.